### PR TITLE
feat(TBD-13123): K8s on Livy and Datatap

### PR DESCRIPTION
**Pull request by AlixMetivier-zz**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java [hunk 1]
```diff
@@ -740,7 +904,8 @@ protected void addFields() {
         addConnectionFields(bigComposite);
         addWebHDFSEncryptionFields(bigComposite);
         addAuthenticationFields(bigComposite);
-      
+        
+        addKubernetesField();
```
Comment (lbourgeois) [inline main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java:908]:
```suggestion
        addKubernetesFields();
```

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java [hunk 2]
```diff
@@ -938,6 +1144,58 @@ private List<String> getClusterTypes() {
...
+            if (runSubmitModes != null) {
+            	bucketCloudLabelNames = runSubmitModes.stream().map(mode -> {
+                    return mode.getLabel();
+                }).collect(Collectors.toList());
+            }
+        }
+        return bucketCloudLabelNames;
+    }
+    
+    private List<String> getK8sSubmitMode() {
```
Comment (lbourgeois) [inline main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java:1161]:
```suggestion
    private List<String> getK8sSubmitModes() {
```

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java [hunk 3]
```diff
@@ -938,6 +1144,58 @@ private List<String> getClusterTypes() {
...
+        List<String> providerLableNames = new ArrayList<String>();
+        if (sparkDistribution != null) {
+            List<EKubernetesSubmitMode> supportCloudProviders = sparkDistribution.getK8sRunSubmitMode();
```
Comment (lbourgeois) [inline main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java:1164]:
```suggestion
            List<EKubernetesSubmitMode> supportCloudProviders = sparkDistribution.getK8sRunSubmitModes();
```

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EKubernetesAzureCredentials.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+// along with this program; if not, write to Talend SA
+// 9 rue Pages 92150 Suresnes, France
+//
+// ============================================================================
+package org.talend.hadoop.distribution.constants.databricks;
+
+import org.talend.hadoop.distribution.i18n.Messages;
+public enum EKubernetesAzureCredentials {
+
+	SECRET(Messages.getString("EKubernetesAzureCredentials.SECRET"), "SECRET");
```
[the remark this answers is not in the document]
Comment (AlixMetivier) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EKubernetesAzureCredentials.java:18]: I expect at some point to have multiple values for this

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EKubernetesS3Credentials.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+// along with this program; if not, write to Talend SA
+// 9 rue Pages 92150 Suresnes, France
+//
+// ============================================================================
+package org.talend.hadoop.distribution.constants.databricks;
+
+import org.talend.hadoop.distribution.i18n.Messages;
+public enum EKubernetesS3Credentials {
+
+	ACCESSANDSECRET(Messages.getString("EKubernetesS3Credentials.ACCESSANDSECRET"), "ACCESSANDSECRET");
```
[the remark this answers is not in the document]
Comment (AlixMetivier) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EKubernetesS3Credentials.java:18]: same, at some point we might have more

**Review by lbourgeois — APPROVED** [review]
LGTM

**Pull request merged**